openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -235,47 +211,76 @@ def get_cohort(user, course_key, assign=True):
         if not assign:
             return None
 
-    choices = course.auto_cohort_groups
-    if len(choices) > 0:
-        # Randomly choose one of the auto_cohort_groups, creating it if needed.
-        group_name = local_random().choice(choices)
+    course_cohorts = get_course_cohorts(course, assignment_type=CourseUserGroup.RANDOM)
+    if course_cohorts:
+        course_cohort = local_random().choice(course_cohorts)
     else:
-        # Use the "default cohort".
-        group_name = DEFAULT_COHORT_NAME
+        course_cohort, __ = CourseUserGroup.objects.get_or_create(
+            course_id=course_key,
+            group_type=CourseUserGroup.COHORT,
+            name=DEFAULT_COHORT_NAME,
+            defaults={'assignment_type': CourseUserGroup.RANDOM}
+        )
 
-    group, __ = CourseUserGroup.objects.get_or_create(
-        course_id=course_key,
-        group_type=CourseUserGroup.COHORT,
-        name=group_name
+    user.course_groups.add(course_cohort)
+
+    return course_cohort
+
+
+def migrate_cohort_settings(course):
```

Comment:
* Actually all the cohort related config settings will be moved to instructor dashboard under the cohort management section. So we need to migrate the existing settings from modulestore to MySQL.
* After we are done with all cohort specific migrations, we will remove the cohort related config settings from studio. So there will be no way to change cohort settings in modulestore.
* During course export we don't cater cohort settings.